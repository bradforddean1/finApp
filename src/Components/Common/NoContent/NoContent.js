import React from "react";
import "./NoContent.scss";
import PropTypes from "prop-types";

/**
 * Renders standarized content when pendig user input.
 * @component
 */
function NoContent(props) {
	const { children } = props;
	return <div className="NoContent">{children}</div>;
}

NoContent.propTypes = {
	children: PropTypes.string.isRequired,
};

export default NoContent;
