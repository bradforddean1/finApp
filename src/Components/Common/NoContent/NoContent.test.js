import React from "react";
import ReactDOM from "react-dom";
import NoContent from "./NoContent";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("NoContent Component", () => {
	it("Renders", () => {
		const div = document.createElement("div");
		ReactDOM.render(<NoContent />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it("Renders the Default State", () => {
		const wrapper = shallow(<NoContent />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("Renders given props", () => {
		const wrapper = shallow(<NoContent>Test</NoContent>);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
