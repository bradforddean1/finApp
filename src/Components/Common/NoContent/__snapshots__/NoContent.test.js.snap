// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoContent Component Renders given props 1`] = `
<div
  className="NoContent"
>
  Test
</div>
`;

exports[`NoContent Component Renders the Default State 1`] = `
<div
  className="NoContent"
/>
`;
